code/game/objects/effects/decals/Cleanable/fuel.dm

```diff
@@ -6,66 +6,74 @@
-				has_spread = 1
-				break
+	var/has_spread = 0
+	//Be absorbed by any other liquid fuel in the tile.
+	for(var/obj/effect/decal/cleanable/liquid_fuel/other in newLoc)
+		if(other != src)
+			other.amount += src.amount
+			other.Spread()
+			has_spread = 1
+			break
 
+	. = ..()
+	if(!has_spread)
+		Spread()
+	else
+		qdel(src)
+
+/obj/effect/decal/cleanable/liquid_fuel/proc/Spread(exclude=list())
+	//Allows liquid fuels to sometimes flow into other tiles.
+	if(amount < 15) return //lets suppose welder fuel is fairly thick and sticky. For something like water, 5 or less would be more appropriate.
+	var/turf/simulated/S = loc
+	if(!istype(S)) return
+	for(var/d in cardinal)
+		var/turf/simulated/target = get_step(src,d)
+		var/turf/simulated/origin = get_turf(src)
+		if(origin.CanPass(null, target, 0, 0) && target.CanPass(null, origin, 0, 0))
+			var/obj/effect/decal/cleanable/liquid_fuel/other_fuel = locate() in target
+			if(other_fuel)
+				other_fuel.amount += amount*0.25
+				if(!(other_fuel in exclude))
+					exclude += src
+					other_fuel.Spread(exclude)
+			else
+				new/obj/effect/decal/cleanable/liquid_fuel(target, amount*0.25,1)
+			amount *= 0.75
+
+
+/obj/effect/decal/cleanable/liquid_fuel/flamethrower_fuel
+	icon_state = "mustard"
+	anchored = 0
+	var/turf/origin
+	New(newLoc, amt = 1, d = 0, var/turf/_origin)
+		origin = _origin
+		set_dir(d) //Setting this direction means you won't get torched by your own flamethrower.
 		. = ..()
-		if(!has_spread)
-			Spread()
-		else
-			qdel(src)
 
-	proc/Spread(exclude=list())
-		//Allows liquid fuels to sometimes flow into other tiles.
-		if(amount < 15) return //lets suppose welder fuel is fairly thick and sticky. For something like water, 5 or less would be more appropriate.
-		var/turf/simulated/S = loc
-		if(!istype(S)) return
-		for(var/d in cardinal)
-			var/turf/simulated/target = get_step(src,d)
-			var/turf/simulated/origin = get_turf(src)
-			if(origin.CanPass(null, target, 0, 0) && target.CanPass(null, origin, 0, 0))
-				var/obj/effect/decal/cleanable/liquid_fuel/other_fuel = locate() in target
-				if(other_fuel)
-					other_fuel.amount += amount*0.25
-					if(!(other_fuel in exclude))
-						exclude += src
-						other_fuel.Spread(exclude)
-				else
-					new/obj/effect/decal/cleanable/liquid_fuel(target, amount*0.25,1)
-				amount *= 0.75
+/obj/effect/decal/cleanable/liquid_fuel/flamethrower_fuel/Spread()
+	//The spread for flamethrower fuel is much more precise, to create a wide fire pattern.
+	var/turf/simulated/S = loc
+	if(amount < 0.1)
+		S.hotspot_expose((T20C*2) + 380,500)
+		return
 
+	if(!istype(S)) return
 
-	flamethrower_fuel
-		icon_state = "mustard"
-		anchored = 0
-		New(newLoc, amt = 1, d = 0)
-			set_dir(d) //Setting this direction means you won't get torched by your own flamethrower.
-			. = ..()
+	for(var/d in list(turn(dir,90),turn(dir,-90), dir))
+		var/turf/simulated/O = get_step(S,d)
+		if (O == origin)
+			continue //No torching the user
```

Comment:
If we have this check, why do we need to set dir in New, not in Initialize?

Reply:
dir is for the direction it travels, it moves away from the user.
but this check is to prevent fuel from spreading back onto the user's tile